### DISCUSSION TITLE
Key columns "userId" and "id" are of incompatible types: text and uuid.

### DISCUSSION THREAD

**Discussion by ngregrichardson** [prisma/prisma#17234] [Q&A] [answered]
I have this basic schema:

```
model User {
  id                       String                    @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  devices                  UserDevice[]
}

model UserDevice {
  id                         String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  user                       User     @relation(fields: [userId], references: [id])
  userId                     String   @db.Uuid
}
```

and when I try to run `prisma migrate dev --name init` it gives me:

```
Migration `20230111050124_init` failed to apply cleanly to the shadow database. 
Error:
db error: ERROR: foreign key constraint "UserDevice_userId_fkey" cannot be implemented
DETAIL: Key columns "userId" and "id" are of incompatible types: text and uuid.
   0: sql_migration_connector::validate_migrations
           with namespaces=None
             at migration-engine\connectors\sql-migration-connector\src\lib.rs:297
   1: migration_core::state::DevDiagnostic
             at migration-engine\core\src\state.rs:264
```

I haven't had problems with stuff like this when I used Prisma, so I need clarification on if I'm missing something.

Thanks!

**Comment by nurul3101** [comment]
Hello @ngregrichardson 👋 

Which prisma version are you using?

I replicated this on my end with version `4.8.1` and it worked as expected for me.

**Reply by ngregrichardson** [reply]
I get this from the CLI

```
$ prisma --version
Environment variables loaded from .env
prisma                  : 4.8.1
@prisma/client          : Not found
Current platform        : windows
Query Engine (Node-API) : libquery-engine d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules\.pnpm\@prisma+engines@4.8.1\node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules\.pnpm\@prisma+engines@4.8.1\node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules\.pnpm\@prisma+engines@4.8.1\node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt d6e67a83f971b175a593ccc12e15c4a757f93ffe (at node_modules\.pnpm\@prisma+engines@4.8.1\node_modules\@prisma\engines\prisma-fmt-windows.exe)
Format Wasm             : @prisma/prisma-fmt-wasm 4.8.0-61.d6e67a83f971b175a593ccc12e15c4a757f93ffe
Default Engines Hash    : d6e67a83f971b175a593ccc12e15c4a757f93ffe
Studio                  : 0.479.0
Preview Features        : postgresqlExtensions
```

and this in `package.json`

```json
"devDependencies": {
    "prisma": "^4.8.1",
}
```

Could it be something in the way my database is set up?

**Reply by ngregrichardson** [reply]
Ah, found the solution. It turns out I had a previous migration attempt that had failed but there was still a migration file in `prisma/migrations`. Clearing the migrations folder fixed my problem. Thanks for your response!

**Reply by moejavascriptt** [reply]
thank you so much bro this worked for me.

**Comment by ngregrichardson** [comment, marked as answer]
Deleted a previous failed migration in `prisma/migrations`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
